js/widgets/forms/textinput.js

```diff
@@ -106,57 +108,40 @@ return $.widget( "mobile.textinput", {
 		});
 	},
 
-	_enhance: function() {
-		var elementClasses = [];
-
-		if ( this.isTextarea ) {
-			elementClasses.push( "ui-input-text" );
-		}
-
-		if ( this.isTextarea || this.isRange ) {
-			elementClasses.push( "ui-shadow-inset" );
+	_addClasses: function() {
+		if ( this._searchIcon ) {
+			this._addClass( this._searchIcon, "ui-textinput-search-icon" );
```

Comment:
```this._searchIcon``` *is* a span: https://github.com/jquery/jquery-mobile/pull/8059/files#diff-22c98d48e778833c3a32e3718fde9a43R126